# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/nsDragServiceProxy.cpp
# Commit: 0d0f19a4db70
# Full Hash: 0d0f19a4db70e734724b99c959148fddef94811e
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2022-08-03 03:42:48
# Regressor Bug: 1781129
# File Overlap Count: 1
# Description:
#   Bug 1781129 - Part 1: Use BigBuffer for IPCDataTransfer, r=edgar
#   
#   The IPCDataTransfer type is used to transfer Clipboard/Drag & Drop payloads
#   over IPC to allow them to be written to or read from the relevant system
#   interfaces. Previously, the system which was used was somewhat complex, and
# ==============================================================================

diff -r 65a0b02c1909 -r 0d0f19a4db70 widget/nsDragServiceProxy.cpp
--- a/widget/nsDragServiceProxy.cpp	Tue Aug 02 21:34:06 2022 +0300
+++ b/widget/nsDragServiceProxy.cpp	Tue Aug 02 18:09:40 2022 +0000
@@ -80,8 +80,8 @@
         }
 
         mozilla::Unused << child->SendInvokeDragSession(
-            dataTransfers, aActionType, Some(std::move(surfaceData)), stride,
-            dataSurface->GetFormat(), dragRect, principal, csp, csArgs,
+            std::move(dataTransfers), aActionType, Some(std::move(surfaceData)),
+            stride, dataSurface->GetFormat(), dragRect, principal, csp, csArgs,
             mSourceWindowContext);
         StartDragSession();
         return NS_OK;
@@ -90,8 +90,9 @@
   }
 
   mozilla::Unused << child->SendInvokeDragSession(
-      dataTransfers, aActionType, Nothing(), 0, static_cast<SurfaceFormat>(0),
-      dragRect, principal, csp, csArgs, mSourceWindowContext);
+      std::move(dataTransfers), aActionType, Nothing(), 0,
+      static_cast<SurfaceFormat>(0), dragRect, principal, csp, csArgs,
+      mSourceWindowContext);
   StartDragSession();
   return NS_OK;
 }